app/code/Magento/Wishlist/Api/WishlistRepositoryInterface.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Wishlist\Api;
+
+use Magento\Wishlist\Api\Data\WishlistInterface;
+
+/**
+ * Interface WishlistRepositoryInterface
+ * @api
+ * @package Magento\Wishlist\Api
+ */
+interface WishlistRepositoryInterface
+{
+    /**
+     * Get Wishlist by id
+     * @param int $id
+     * @return \Magento\Wishlist\Api\Data\WishlistInterface
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function getById($id): \Magento\Wishlist\Api\Data\WishlistInterface;
+
+    /**
+     * Get list of wishlists by criteria
+     *
+     * @param \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria
+     * @return \Magento\Wishlist\Api\Data\WishlistSearchResultsInterface
+     */
+    public function getList(\Magento\Framework\Api\SearchCriteriaInterface $searchCriteria);
+
+    /**
+     * Save wishlist object
+     *
+     * @param Data\WishlistInterface $wishlist
+     * @return \Magento\Wishlist\Api\Data\WishlistInterface
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
+     */
+    public function save(\Magento\Wishlist\Api\Data\WishlistInterface $wishlist): WishlistInterface;
```

Comment:
To make this method more CQS-friendly it may be updated to return only id of the created item, instead of the whole item. Returning void may be a good idea as well, but would greatly limit use cases, as there is no predefined if for wishlist.

Reply:
Changed to returning id

Reply:
I've changed this method to return just the id